src/lib/flight_tasks/tasks/ManualPosition/FlightTaskManualPosition.cpp

```diff
@@ -169,7 +169,13 @@ void FlightTaskManualPosition::_updateSetpoints()
 	// check if an external yaw handler is active and if yes, let it update the yaw setpoints
 	if (_weathervane_yaw_handler != nullptr && _weathervane_yaw_handler->is_active()) {
 		_yaw_setpoint = NAN;
-		_yawspeed_setpoint += _weathervane_yaw_handler->get_weathervane_yawrate();
+
+		// only enable the weathervane to change the yawrate when position lock is active (and thus the pos. sp. are NAN)
+		if (PX4_ISFINITE(_position_setpoint(0)) && PX4_ISFINITE(_position_setpoint(1))) {
```

Comment:
Given the position lock check should this be done after the `_updateXYlock()` call below?

Reply:
Good point, I've moved it.